09_json/n0npax/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"bou.ke/monkey"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainNoArgs(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	args = []string{}
+	main()
+
+	expected := ""
+	actual := buf.String()
+	assert.Equal(t, expected, actual)
+}
+
+func TestMainWithArgs(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	args = []string{"--data", "../../test-data/data.json"}
+	main()
+
+	expected := ""
+	actual := buf.String()
+	assert.Equal(t, expected, actual)
+}
+
+func TestMainCorruptedData(t *testing.T) {
```

Comment:
Should the function named be called `TestMainWithIncorrectFilePath` or something similar? as you try to fake an incorrect file. Sorry to be pandatic.

Reply:
`/dev/null` is a file, so the path is fine. `/dev/null` contains just `NULL` and this is not valid puppy data. In my mind the proposed name is confusing.